File: Plugin/app/info/schleichardt/play2/mailplugin/api/Mailer.scala
package info.schleichardt.play2.mailplugin.api

import org.apache.commons.mail.Email
import info.schleichardt.play2.mailplugin.{EmailSendInterceptor, MailPlugin}
import scala.collection.JavaConversions._

object Mailer {
  def send(email: Email, profile: String = ""): String = MailPlugin.instance.send(email, profile)

  def history: Seq[Email] = MailPlugin.instance.mailArchive.clone()

  protected[mailplugin] def historyAsJava: java.util.List[Email] = history

  def setInterceptor(interceptor: EmailSendInterceptor) {
    require(interceptor != null)
    MailPlugin.instance.interceptor = interceptor
  }
}

File: Plugin/app/info/schleichardt/play2/mailplugin/Mailer.java
package info.schleichardt.play2.mailplugin;

import info.schleichardt.play2.mailplugin.interceptors.EmailInterceptor;
import org.apache.commons.mail.Email;
import org.apache.commons.mail.EmailException;
import play.Play;
import scala.collection.JavaConverters;
import scala.collection.Seq;

import java.util.List;

public class Mailer {
    private Mailer() {
    }

    /** send an email with default profile, for example with smtp.user setting */
    public static String send(final Email email) throws EmailException {
        return send(email, "");
    }

    /** send an email with a special profile, for example with smtp.profiles.aprofile.user setting */
    public static String send(final Email email, final String profile) throws EmailException {
        return getMailPluginInstance().send(email, profile);
    }

    public static List<Email> history() {
        final Seq<Email> HistoryAsScalaSeq = getMailPluginInstance().history();
        return JavaConverters.seqAsJavaListConverter(HistoryAsScalaSeq).asJava();
    }

    public static void addInterceptor(final EmailInterceptor interceptor) {
        getMailPluginInstance().addInterceptor(interceptor);
    }

    public static void removeInterceptor(final EmailInterceptor interceptor) {
        getMailPluginInstance().removeInterceptor(interceptor);
    }

    private static MailPlugin getMailPluginInstance() {
        return Play.application().plugin(MailPlugin.class);
    }
}
